### ISSUE TITLE
App Keeps Connecting to Broken Stations

### ISSUE THREAD

**Issue by zo-shin** [segler-alex/RadioDroid#771]
If a broken station is clicked, the app will keep connecting to it.

**Comment by werman** [comment]
Without digging deeper: what's the difference between connecting to not working stream and troubles with a network, we don't know whose fault is it. So this looks like a correct behavior.

**Comment by zo-shin** [comment]
In version 0.7x, the app shows an error at once. But now it keeps connecting till users stop it.

**Comment by werman** [comment]
Probably we can stop reconnecting if we get some certain http error codes. Should help in most cases.

**Comment by werman** [comment]
Yes, we don't handle error codes properly. I'll refactor the data source and player wrapper to show errors on http error codes and invalid mime types. 

And will get rid of two timeouts, leaving only "Retry timeout" but making it more robust.
